lib/govuk_index/publishing_event_worker.rb

```diff
@@ -56,17 +48,48 @@ def process_action(actions, routing_key, payload)
       else
         logger.info("#{routing_key} -> SKIPPED #{presenter.base_path} #{presenter.type}")
       end
+
+    # Rescuing as we don't want to retry this class of error
+    rescue MissingBasePath => e
+      return if DOCUMENT_TYPES_WITHOUT_BASE_PATH.include?(payload["document_type"])
+      GovukError.notify(e, extra: { message_body: payload })
+        # Unpublishing messages for something that does not exist may have been
+        # processed out of order so we don't want to notify errbit but just allow
+        # the process to continue
+    rescue NotFoundError
+      logger.info("#{payload['base_path']} could not be found.")
+      Services.statsd_client.increment('govuk_index.not-found-error')
+    rescue UnknownDocumentTypeError
+      logger.info("#{payload['document_type']} document type is not known.")
+      Services.statsd_client.increment('govuk_index.unknown-document-type')
     end
 
-    def process_response(response)
-      # we are only expecting to process a single message at a time
-      if response['items'].count > 1
-        Services.statsd_client.increment('govuk_index.elasticsearch.multiple_responses')
-        raise MultipleMessagesInElasticsearchResponse
+    def process_responses(responses, messages)
+      messages_with_error = []
+
+      responses['items'].zip(messages).each do |response, message|
+        messages_with_error << message unless valid_response?(response)
       end
 
-      item = response['items'].first
-      action_type, details = *item.first # item is a hash with a single key, value pair
+      if messages_with_error.any?
+        if messages_with_error.count == messages.count
+          # automatically retry
```

Comment:
I don't think we should have a special case for complete failure. It would be simpler if the partial retry handled everything. Especially since the partial retry code is otherwise not expected to execute in production so we won't notice if there are any problems with it.

Reply:
The reason for this differentiation was that the partial retry loses the sidekiq retry implementation. I might have a look at how retry works and see if I can do partial retry with backoff and limited count of attempts.